base_layer/core/src/transactions/transaction_components/encrypted_openings_x.rs

```diff
@@ -0,0 +1,227 @@
+
+use super::EncryptedOpeningsKey;
+use crate::transactions::{tari_amount::MicroTari, TransactionKdfDomain};
+
+const VALUE_SIZE: usize = 8;
+const KEY_SIZE: usize = 32;
+const TAG_SIZE: usize = 16;
+const SIZE: usize = VALUE_SIZE + KEY_SIZE + TAG_SIZE + size_of::<XNonce>();
+const BORSH_64: usize = 64;
+const BORSH_X: usize = SIZE - BORSH_64;
+
+/// Encrypted openings for the extended-nonce variant XChaCha20-Poly1305 encryption
+#[derive(
+    Debug, Copy, Clone, Deserialize, Serialize, PartialEq, Eq, Hash, BorshSerialize, BorshDeserialize, Zeroize,
+)]
+pub struct EncryptedOpeningsX {
+    #[serde(with = "tari_utilities::serde::hex")]
+    data_1: [u8; BORSH_64],
+    #[serde(with = "tari_utilities::serde::hex")]
+    data_2: [u8; BORSH_X],
+}
+
+impl EncryptedOpeningsX {
+    /// Custom convert `EncryptedOpenings` to bytes
+    pub fn from_bytes(bytes: &[u8]) -> Result<Self, EncryptedOpeningsErrorX> {
+        if bytes.len() != SIZE {
+            return Err(EncryptedOpeningsErrorX::IncorrectLength(format!(
+                "Expected {} bytes, got {}",
+                SIZE,
+                bytes.len()
+            )));
+        }
+        let mut data_1: [u8; BORSH_64] = [0u8; BORSH_64];
+        data_1.copy_from_slice(
+            bytes
+                .get(..BORSH_64)
+                .ok_or_else(|| EncryptedOpeningsErrorX::IncorrectLength("Out of bounds 'data_1'".to_string()))?,
+        );
+        let mut data_2: [u8; BORSH_X] = [0u8; BORSH_X];
+        data_2.copy_from_slice(
+            bytes
+                .get(BORSH_64..SIZE)
+                .ok_or_else(|| EncryptedOpeningsErrorX::IncorrectLength("Out of bounds 'data_2'".to_string()))?,
+        );
+        Ok(Self { data_1, data_2 })
+    }
+
+    /// Custom convert `EncryptedOpenings` to byte vector
+    pub fn as_byte_vector(&self) -> Vec<u8> {
+        let mut bytes = Vec::with_capacity(SIZE);
+        bytes.extend_from_slice(&self.data_1);
+        bytes.extend_from_slice(&self.data_2);
+        bytes
+    }
+}
+
+impl Default for EncryptedOpeningsX {
+    fn default() -> Self {
+        Self {
+            data_1: [0u8; BORSH_64],
+            data_2: [0u8; BORSH_X],
+        }
+    }
+}
+
+/// EncryptedOpenings errors
+#[derive(Debug, Error)]
+pub enum EncryptedOpeningsErrorX {
+    #[error("Encryption failed: {0}")]
+    EncryptionFailed(Error),
+    #[error("Conversion failed: {0}")]
+    ByteArrayError(#[from] ByteArrayError),
+    #[error("Incorrect length: {0}")]
+    IncorrectLength(String),
+}
+
+// Chacha error is not StdError compatible
+impl From<Error> for EncryptedOpeningsErrorX {
+    fn from(err: Error) -> Self {
+        Self::EncryptionFailed(err)
+    }
+}
+
+impl EncryptedOpeningsX {
+    const TAG: &'static [u8] = b"TARI_AAD_VALUE_AND_MASK";
+
+    /// Encrypt the value and mask (with fixed length) using XChaCha20-Poly1305 with a secure random nonce
+    pub fn encrypt_openings(
+        encryption_key: &PrivateKey,
+        commitment: &Commitment,
```

Comment:
With the use of a secure random nonce, there's no need to use the commitment in the internal key derivation. You can certainly keep it for consistency if you like, since it doesn't hurt.